Kitodo/src/test/java/org/kitodo/production/process/NewspaperProcessesGeneratorIT.java

```diff
@@ -0,0 +1,118 @@
+import org.kitodo.FileLoader;
+import org.kitodo.MockDatabase;
+import org.kitodo.NewspaperCourse;
+import org.kitodo.SecurityTestUtils;
+import org.kitodo.TreeDeleter;
+import org.kitodo.config.ConfigCore;
+import org.kitodo.config.enums.KitodoConfigFile;
+import org.kitodo.config.enums.ParameterCore;
+import org.kitodo.data.database.beans.Process;
+import org.kitodo.production.model.bibliography.course.Course;
+import org.kitodo.production.model.bibliography.course.Granularity;
+import org.kitodo.production.services.ServiceManager;
+import org.kitodo.production.services.data.ProcessService;
+import org.kitodo.production.services.file.FileService;
+
+public class NewspaperProcessesGeneratorIT {
+    private static final FileService fileService = new FileService();
+    private static final ProcessService processService = ServiceManager.getProcessService();
+
+    private static final String firstProcess = "First process";
+    private static final File script = new File(ConfigCore.getParameter(ParameterCore.SCRIPT_CREATE_DIR_META));
+
+    @Rule
+    public ExpectedException expectedEx = ExpectedException.none();
+
+    /**
+     * The test environment is being set up.
+     *
+     * @throws Exception
+     *             if that does not work
+     */
+    @BeforeClass
+    public static void setUp() throws Exception {
+        if (!SystemUtils.IS_OS_WINDOWS) {
+            ExecutionPermission.setExecutePermission(script);
+        }
+        FileLoader.createConfigProjectsFileForCalendarHierarchyTests();
+        MockDatabase.startNode();
+        MockDatabase.insertProcessesFull();
+        MockDatabase.insertProcessForCalendarHierarchyTests();
+        MockDatabase.setUpAwaitility();
+        fileService.createDirectory(URI.create(""), "1");
+        SecurityTestUtils.addUserDataToSecurityContext(ServiceManager.getUserService().getById(1), 1);
+        while (Objects.isNull(processService.findByTitle(firstProcess))) {
+            Thread.sleep(256);
+        }
+    }
+
+    /**
+     * The test environment is cleaned up and the database is closed.
+     */
+    @AfterClass
+    public static void tearDown() throws Exception {
+        MockDatabase.stopNode();
+        MockDatabase.cleanDatabase();
+        KitodoConfigFile.PROJECT_CONFIGURATION.getFile().delete();
+    }
+
+    /**
+     * Perform the test. A long-running task is simulated: Progress and number
+     * of steps are queried and the next step is performed.
+     */
+    @Test
+    public void shouldGenerateNewspaperProcesses() throws Exception {
+        Process completeEdition = ServiceManager.getProcessService().getById(10);
+        Course course = NewspaperCourse.getCourse();
+        course.splitInto(Granularity.DAYS);
+        NewspaperProcessesGenerator underTest = new NewspaperProcessesGenerator(completeEdition, course);
+        while (underTest.getProgress() < underTest.getNumberOfSteps()) {
+            underTest.nextStep();
+        }
+        cleanUp();
+    }
```

Comment:
There is no Assert in the Test :(

Reply:
I added an assertion.

Reply:
Sorry, this is not an acceptable Assertion.
No output is testet, only the final progress is testet but no steps in between. What about "expected results" "Actual results"?
What about testing the data structure? generated Processes, generated metadata...